ts/parts-more/Polar.ts

```diff
@@ -352,74 +351,70 @@ if (seriesTypes.spline) {
 H.addEvent(Series as any, 'afterTranslate', function (
     this: Highcharts.PolarSeries
 ): void {
-    var chart = this.chart,
-        points,
-        i;
+    const series = this;
+    const chart = series.chart;
 
-    if (chart.polar && this.xAxis) {
+    if (chart.polar && series.xAxis) {
 
         // Prepare k-d-tree handling. It searches by angle (clientX) in
         // case of shared tooltip, and by two dimensional distance in case
         // of non-shared.
-        this.kdByAngle = chart.tooltip && chart.tooltip.shared;
-        if (this.kdByAngle) {
-            this.searchPoint = this.searchPointByAngle;
+        series.kdByAngle = chart.tooltip && chart.tooltip.shared;
+        if (series.kdByAngle) {
+            series.searchPoint = series.searchPointByAngle;
         } else {
-            this.options.findNearestPointBy = 'xy';
+            series.options.findNearestPointBy = 'xy';
         }
 
         // Postprocess plot coordinates
-        if (!this.preventPostTranslate) {
-            points = this.points;
-            i = points.length;
+        if (!series.preventPostTranslate) {
+            const points = series.points;
+
+            let i = points.length;
 
             while (i--) {
                 // Translate plotX, plotY from angle and radius to true plot
                 // coordinates
-                this.toXY(points[i]);
+                series.toXY(points[i]);
 
                 // Treat points below Y axis min as null (#10082)
                 if (
                     !chart.hasParallelCoordinates &&
-                    !this.yAxis.reversed &&
-                    (points[i].y as any) < (this.yAxis.min as any)
+                    !series.yAxis.reversed &&
+                    (points[i].y as any) < (series.yAxis.min as any)
                 ) {
                     points[i].isNull = true;
                 }
             }
         }
 
         // Perform clip after render
-        if (!this.hasClipCircleSetter) {
```

Comment:
I think this `if` should stay, otherwise we have leak here: http://jsfiddle.net/BlackLabel/em2hbrwn/ 

Click on legend multiple times and observe results in legend. The same happens on window's resize.